src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -673,27 +652,55 @@ private static (string, MultipartContent) GetMultipartContent(string contentSour
         return (sb.ToString(), multipartContent);
     }
 
-    /// <summary>Client context containing information pertaining to a single worker.</summary>
-    private sealed class ClientContext
+    /// <summary>Client context containing information pertaining to a single request.</summary>
+    private sealed class RequestContext
     {
         private readonly Random _random;
+        private readonly HttpClient _client;
+        private readonly double _cancellationProbability;
 
-        public ClientContext(HttpClient httpClient, int taskNum, int seed)
+        public RequestContext(HttpClient httpClient, Random random, int taskNum, double cancellationProbability)
         {
-            _random = new Random(Combine(seed, taskNum)); // derived from global seed and worker number
+            _random = random;
+            _client = httpClient;
+            _cancellationProbability = cancellationProbability;
+
             TaskNum = taskNum;
-            HttpClient = httpClient;
+            IsCancellationRequested = false;
+        }
+
+        public int TaskNum { get; }
+        public bool IsCancellationRequested { get; set; }
 
-            // deterministic hashing copied from System.Runtime.Hashing
-            int Combine(int h1, int h2)
+        // HttpClient.SendAsync() wrapper that wires randomized cancellation
+        public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompletionOption httpCompletion = HttpCompletionOption.ResponseContentRead, CancellationToken? token = null)
+        {
+            if (token != null)
             {
-                uint rol5 = ((uint)h1 << 5) | ((uint)h1 >> 27);
-                return ((int)rol5 + h1) ^ h2;
+                // user-supplied cancellation token overrides random cancellation
+                return await _client.SendAsync(request, httpCompletion, token.Value);
             }
```

Comment:
Rather than overriding, you can instead combine, e.g. instead of:
```C#
using (var cts = new CancellationTokenSource())
```
do:
```C#
using (CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(token))
```

Reply:
My thinking here is that passing an explicit token designates intent to test a specific cancellation scenario, thus mixing in randomized cancellation might complicate things.

Reply:
Ok.